packages/headless/src/features/generated-answer/generated-answer-insight-analytics-actions.ts

```diff
@@ -51,46 +52,50 @@ export const logRephraseGeneratedAnswer = (
 export const logOpenGeneratedAnswerSource = (
   citationId: string
 ): InsightAction =>
-  makeInsightAnalyticsAction({
-    prefix: 'analytics/generatedAnswer/openAnswerSource',
-    __legacy__getBuilder: (client, state) => {
-      const generativeQuestionAnsweringId =
-        generativeQuestionAnsweringIdSelector(state);
-      const citation = citationSourceSelector(state, citationId);
+  makeInsightAnalyticsActionFactory(SearchPageEvents.openGeneratedAnswerSource)(
+    {
+      prefix: 'analytics/generatedAnswer/openAnswerSource',
+      __legacy__getBuilder: (client, state) => {
+        const generativeQuestionAnsweringId =
+          generativeQuestionAnsweringIdSelector(state);
+        const citation = citationSourceSelector(state, citationId);
 
-      if (!generativeQuestionAnsweringId || !citation) {
-        return null;
-      }
-      return client.logOpenGeneratedAnswerSource(
-        {
-          generativeQuestionAnsweringId,
-          permanentId: citation.permanentid,
-          citationId: citation.id,
-        },
-        getCaseContextAnalyticsMetadata(state.insightCaseContext)
-      );
-    },
-    analyticsType: 'Qna.CitationClick',
-    analyticsPayloadBuilder: (state): Qna.CitationClick => {
-      const generativeQuestionAnsweringId =
-        generativeQuestionAnsweringIdSelector(state);
-      return {
-        answer: {
-          id: generativeQuestionAnsweringId!,
-          type: 'CRGA',
-        },
-        citation: {
-          id: citationId,
-        },
-      };
-    },
-  });
+        if (!generativeQuestionAnsweringId || !citation) {
+          return null;
+        }
+        return client.logOpenGeneratedAnswerSource(
+          {
+            generativeQuestionAnsweringId,
+            permanentId: citation.permanentid,
+            citationId: citation.id,
+          },
+          getCaseContextAnalyticsMetadata(state.insightCaseContext)
+        );
+      },
+      analyticsType: 'Qna.CitationClick',
+      analyticsPayloadBuilder: (state): Qna.CitationClick => {
+        const generativeQuestionAnsweringId =
+          generativeQuestionAnsweringIdSelector(state);
+        return {
+          answer: {
+            id: generativeQuestionAnsweringId!,
```

Comment:
In the pattern we currently have for the actions that log analytics using the event protocol, I saw that we always try to return a event.

I think now that these events are being validated, we just need to send the events, if some payload is missing, the event will just not be valid and it will be rejected.
Sending null will kinda silence the sent of the event and will not add much of a value in my opinion.